src/App.jsx

```diff
@@ -3,69 +3,134 @@ import './App.css'
 import MovieList from './MovieList'
 import Header from './Header'
 import Footer from './Footer'
-
+import SideBar from './SideBar'
 
 const App = () => {
   const [loadedMovies, setLoadedMovies] = useState([]);
   const [page, setPage] = useState(1); //Page number of next page to load
-  const baseURL = "https://image.tmdb.org/t/p/w500";
   const [searchTerm, setSearchTerm] = useState('');
+  const [genre, setGenre] = useState('');
+  const [apiFunctionCall, setApiFunctionCall] = useState('now_playing');
+  const [sortAttribute, setSortAttribute] = useState('');
+  let favoritedList = [];
+  let watchedList = [];
 
-  const generalFetchMovies = async (fxn, prevMovieList, pages, searchTerm) => {
+  const generalFetchMovies = async (fxn, prevMovieList) => {
     const apiKey = import.meta.env.VITE_API_KEY;
     let response = {};
+    let url = '';
 
     switch (fxn) {
       case 'now_playing':
-        response = await fetch(`https://api.themoviedb.org/3/movie/now_playing?language=en-US&page=${pages}&api_key=${apiKey}`);
+        response = await fetch(`https://api.themoviedb.org/3/movie/now_playing?language=en-US&page=${page}&api_key=${apiKey}`);
         break;
       case 'search':
         response = await fetch(`https://api.themoviedb.org/3/search/movie?query=${searchTerm}&include_adult=false&language=en-US&page=1&api_key=${apiKey}`);
         break;
-      default: response = {};
-
+      case 'by_genre':
+        response = await fetch(`https://api.themoviedb.org/3/discover/movie?with_genres=${genre}&language=en-US&page=${page}&api_key=${apiKey}`);
+        break;
+      case 'sort_by_attribute':
+        if (genre === ''){
+          url = `https://api.themoviedb.org/3/discover/movie?include_adult=false&include_video=true&language=en-US&page=${page}&sort_by=${sortAttribute}&api_key=${apiKey}`;
+        } else {
+          url = `https://api.themoviedb.org/3/discover/movie?include_adult=false&with_genres=${genre}&language=en-US&page=${page}&sort_by=${sortAttribute}&api_key=${apiKey}`;
+        }
+        response = await fetch(url);
+        break;
+      default:
+        response = response = await fetch(`https://api.themoviedb.org/3/movie/now_playing?language=en-US&${page}=1&api_key=${apiKey}`);
```

Comment:
- The URIs actually have a lot in common! You can use the [URL library](https://developer.mozilla.org/en-US/docs/Web/API/URL) to generate the links programmatically instead of using the string templates here.

- You likely would want to move this function definition outside of the component. That would make it so it isn't redefined on every render. Also it would separate some concerns and add more readability to the code